src/contracts/SavingCircles.sol

```diff
@@ -185,33 +207,33 @@ contract SavingCircles is ISavingCircles, ReentrancyGuard, OwnableUpgradeable {
   }
 
   /// @inheritdoc ISavingCircles
-  function withdrawableBy(uint256 _id) external view override returns (address) {
+  function withdrawableBy(uint256 _id) external view override onlyCommissioned(_id) returns (address) {
     Circle memory _circle = circles[_id];
 
-    if (_isDecommissioned(_circle)) revert NotCommissioned();
-
     return _circle.members[_circle.currentIndex];
   }
 
   /**
    * @dev Make a withdrawal from a specified circle
    *      A withdrawal must be made by a member of the circle, even if it is for another member.
    */
-  function _withdraw(uint256 _id, address _member) internal {
-    Circle storage _circle = circles[_id];
+  function _withdraw(uint256 _id, address _member) internal onlyMember(_id) {
+    Circle memory _circle = circles[_id];
 
-    if (!isMember[_id][msg.sender]) revert NotMember();
     if (!_withdrawable(_id)) revert NotWithdrawable();
     if (_circle.members[_circle.currentIndex] != _member) revert NotWithdrawable();
     if (_circle.currentIndex >= _circle.maxDeposits) revert NotWithdrawable();
 
-    uint256 _withdrawAmount = _circle.depositAmount * (_circle.members.length);
+    //Caching the circle members length to avoid multiple lookups
```

Comment:
same comment about this not being a needed comment. remove or add space after //